src/faction_camp.cpp

```diff
@@ -514,6 +514,96 @@ static bool update_time_fixed( std::string &entry, const comp_list &npc_list,
+        const point &dir,
+        bool &mirror_horizontal,
+        bool &mirror_vertical,
+        int &rotation,
+        const std::string base_error_message,
+        const std::string actor )
+{
+    mirror_horizontal = recipe->has_flag( "MAP_MIRROR_HORIZONTAL" );
+    mirror_vertical = recipe->has_flag( "MAP_MIRROR_VERTICAL" );
+    rotation = 0;
+    std::string dir_string;
+
+    const auto check_rotation = [&]( const std::string & flag, int rotation_value ) {
+        if( recipe->has_flag( flag ) ) {
+            if( rotation != 0 ) {
+                debugmsg( "%s, the blueprint specifies multiple concurrent rotations, which is not supported",
+                          string_format( base_error_message, actor, recipe->get_blueprint() ) );
+                return false;
+            }
+            rotation = rotation_value;
+        }
+        return true;
+    };
+
+    if( !check_rotation( "MAP_ROTATE_90", 1 ) ) {
+        return false;
+    }
+
+    if( !check_rotation( "MAP_ROTATE_180", 2 ) ) {
+        return false;
+    }
+
+    if( !check_rotation( "MAP_ROTATE_270", 3 ) ) {
+        return false;
+    }
+
+    if( dir.x == -1 && dir.y == -1 ) {
+        dir_string = "NW";
+    } else if( dir.x == 0 && dir.y == -1 ) {
+        dir_string = "N";
+    } else if( dir.x == 1 && dir.y == -1 ) {
+        dir_string = "NE";
+    } else if( dir.x == -1 && dir.y == 0 ) {
+        dir_string = "W";
+    } else if( dir.x == 0 && dir.y == 0 ) {
+        dir_string = "";  //  Will result in "hidden" flags that can actually affect the core.
+    } else if( dir.x == 1 && dir.y == 0 ) {
+        dir_string = "E";
+    } else if( dir.x == -1 && dir.y == 1 ) {
+        dir_string = "SW";
+    } else if( dir.x == 0 && dir.y == 1 ) {
+        dir_string = "S";
+    } else if( dir.x == 1 && dir.y == 1 ) {
+        dir_string = "SE";
+    }
+
+    if( recipe->has_flag( "MAP_MIRROR_HORIZONTAL_IF_" + dir_string ) ) {
+        if( mirror_horizontal ) {
+            debugmsg( "%s, the blueprint specifies multiple concurrent horizontal mirroring, which is not supported",
+                      string_format( base_error_message, actor, recipe->get_blueprint() ) );
+            return false;
+        }
+        mirror_horizontal = true;
+    }
+
+    if( recipe->has_flag( "MAP_MIRROR_VERTICAL_IF_" + dir_string ) ) {
+        if( mirror_vertical ) {
+            debugmsg( "%s, the blueprint specifies multiple concurrent vertical mirroring, which is not supported",
+                      string_format( base_error_message, actor, recipe->get_blueprint() ) );
+            return false;
+        }
+        mirror_vertical = true;
+    }
+
+    if( !check_rotation( "MAP_ROTATE_90_IF_" + dir_string, 1 ) ) {
+        return false;
+    }
+
+    if( !check_rotation( "MAP_ROTATE_180_IF_" + dir_string, 2 ) ) {
+        return false;
+    }
+
+    if( !check_rotation( "MAP_ROTATE_270_IF_" + dir_string, 3 ) ) {
+        return false;
+    }
+
+    return true;
+};
```

Comment:
```suggestion
}
```

Reply:
Thanks. You're correct. I was fooled by the error report expanding the second failure point, rather than the one where the rejection actually was ("Error: Path does not exist /home/runner/work/Cataclysm-DDA/Cataclysm-DDA/tests/cata_test").